### PR TITLE
Add Sage Provider

### PR DESCRIPTION
**Pull request by lukecurtis93**

### CODE REVIEW

File: src/Sage/Provider.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     * The separating character for the requested scopes.
+     *
+     * @var string
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [ ];
```
Comment (atymic) [inline src/Sage/Provider.php:25]: no scopes?

File: src/Sage/Provider.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
+    {
+        return array_merge(parent::getTokenFields($code), [
+            'grant_type' => 'authorization_code',
+        ]);
+    }
+}
```
Comment (atymic) [inline src/Sage/Provider.php:81]: newline

File: src/Sage/composer.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "description": "Sage OAuth2 Provider for Laravel Socialite",
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "Luke Curtis",
+            "email": "lukesimoncurtis@gmail.com"
+        }
+    ],
+    "require": {
+        "php": ">=5.5.9",
```
Comment (atymic) [inline src/Sage/composer.json:12]: can drop < 7

File: src/Sage/composer.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        "ext-json": "*",
+        "socialiteproviders/manager": "^3.5"
+    },
+    "autoload": {
+        "psr-4": {
+            "SocialiteProviders\\Sage\\": ""
+        }
+    }
+}
```
Comment (atymic) [inline src/Sage/composer.json:21]: newline :)

**Comment by lukecurtis93** [comment]
> Looks good, just a couple of minor bits :)

Thanks for the review mate all good stuff there dude 👍 I've made the requested changes :)

**Pull request merged**